File: TinyCookbook/Assets/TinySamples/MatchThree/Scripts/Utilities/DestroyAfterDelaySystem.ts

namespace game {

    /**
     * Destroy an entity after a delay of time.
     */
    export class DestroyAfterDelaySystem extends ut.ComponentSystem {
        
        OnUpdate(): void {
            
            let deltaTime = this.scheduler.deltaTime();
            this.world.forEach([ut.Entity, game.DestroyAfterDelay],
                (entity, destroyAfterDelay) => {
                    
                    let destroyDelay = destroyAfterDelay.Delay;
                    destroyDelay -= deltaTime;
                    destroyAfterDelay.Delay = destroyDelay;

                    if (destroyDelay <= 0) {
                        ut.Core2D.TransformService.destroyTree(this.world, entity, true);
                    }
                });
        }
    }
}

File: Library/PackageCache/com.unity.tiny@0.16.1-preview/Samples~/SpawnAndDestroy/Scripts/DestroyAfterDelaySystem.cs
using Unity.Collections;
using Unity.Entities;
using Unity.Tiny.Core;
using Unity.Tiny.Scenes;

namespace SpawnAndDestroy
{
    public class DestroyAfterDelaySystem : ComponentSystem
    {
        protected override void OnUpdate()
        {
            var deltaTime = World.TinyEnvironment().frameDeltaTime;
            var entitiesToUnload = new NativeList<Entity>(Allocator.Temp);

            Entities.ForEach((Entity entity, ref DestroyAfterDelay destroyAfterDelay) =>
            {
                destroyAfterDelay.Timer += deltaTime;
                if (destroyAfterDelay.Timer > destroyAfterDelay.DestroyDelay)
                {
                    entitiesToUnload.Add(entity);
                }
            });

            // Unload scenes outside the ForEach since structural changes cannot be made inside a ForEach
            for (int i = 0; i < entitiesToUnload.Length; i++)
                SceneService.UnloadSceneInstance(entitiesToUnload[i]);
            entitiesToUnload.Dispose();
        }
    }
}
